packages/amplify-graphql-transformer-core/src/graphql-api.ts

```diff
@@ -288,20 +310,39 @@ export class GraphQLApi extends GraphqlApiBase implements GraphQLAPIProvider {
     return true;
   }
 
-  private setupLogConfig(config?: LogConfig) {
-    if (!config) {
-      return undefined;
-    }
-    const role = new Role(this, 'ApiLogsRole', {
+  /**
+   * Set up the log configuration for the GraphQL API
+   *
+   * Default values are:
+   * - excludeVerboseContent: true
+   * - fieldLogLevel: NONE
+   */
+  private setupLogConfig(logging?: true | LogConfig): CfnGraphQLApi.LogConfigProperty | undefined {
+    if (!logging) return undefined;
+
+    const defaultExcludeVerboseContent = true;
+    const defaultFieldLogLevel = FieldLogLevel.NONE;
+
+    const excludeVerboseContent =
+      logging === true || (typeof logging === 'object' && Object.keys(logging).length === 0)
+        ? defaultExcludeVerboseContent
+        : logging.excludeVerboseContent ?? defaultExcludeVerboseContent;
+
+    const fieldLogLevel =
+      logging === true || (typeof logging === 'object' && Object.keys(logging).length === 0)
+        ? defaultFieldLogLevel
+        : logging.fieldLogLevel ?? defaultFieldLogLevel;
+
+    const apiLogsRole = new Role(this, 'ApiLogsRole', {
```

Comment:
Role creation subject to change as we are considering a more restrictive policy.